op-service/sources/batching/bound.go

```diff
@@ -17,21 +18,27 @@ var (
 )
 
 type BoundContract struct {
-	abi  *abi.ABI
-	addr common.Address
+	abi    *abi.ABI
+	addr   common.Address
+	filter bind.ContractFilterer
 }
 
-func NewBoundContract(abi *abi.ABI, addr common.Address) *BoundContract {
+func NewBoundContract(abi *abi.ABI, addr common.Address, filter bind.ContractFilterer) *BoundContract {
 	return &BoundContract{
-		abi:  abi,
-		addr: addr,
+		abi:    abi,
+		addr:   addr,
+		filter: filter,
```

Comment:
What is the purpose of this field?  Looks like it is used by nowhere?

Reply:
The `addr` field is used in https://github.com/ethstorage/optimism/blob/15e66c13bcb679211ab548f3d18b83b30b2fc149/op-challenger/game/fault/contracts/delayed_weth.go#L35

Reply:
I guess Qi's referring to the filter field, which seems not being used ?

Reply:
> I guess Qi's referring to the filter field, which seems not being used ?

It's been used at
https://github.com/ethstorage/optimism/blob/15e66c13bcb679211ab548f3d18b83b30b2fc149/op-challenger/game/fault/contracts/faultdisputegame.go#L462C26-L462C36

Reply:
But it's different from the `BoundContract.filter`, no? I still don't find the usage of it.

Reply:
To reuse code generated by abi bindings of the `FilterMove` function, https://github.com/ethstorage/optimism/blob/4b86fd7688b8cd2e6dc99b21f9f09778e213c353/op-e2e/bindings/faultdisputegame.go#L1666
the boundContract should implement a `Filterlogs` function, which requires the boundContract has a `filter` field that implement the `FilterLogs` interface:
https://github.com/ethereum-optimism/op-geth/blob/optimism/accounts/abi/bind/base.go#L436
```
	buff, err := c.filterer.FilterLogs(ensureContext(opts.Context), config)
```

Reply:
it is a private field, right?  why it can be used by external package?

Reply:
Yes, you're right! Since `f.multiCaller` already implements the `bind.ContractFilterer` interface, the filter isn't needed. Initially, I confused the usage of interfaces and attributes.

Fixed in recent commit.